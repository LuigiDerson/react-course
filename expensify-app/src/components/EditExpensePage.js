import React from 'react';

const EditExpensePage = ({match: {params: par}}) => {
    return (
        <div>
            This is from my edit expense Component whit id: {par.id}
        </div>
    );
}

export default EditExpensePage;