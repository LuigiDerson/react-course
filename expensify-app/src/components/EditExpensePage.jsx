import React from "react";

const EditExpensePage = ({ match: { params } }) => (
  <div>This is from my 'edit' expense Component whit id: {params.id}</div>
);

export default EditExpensePage;
