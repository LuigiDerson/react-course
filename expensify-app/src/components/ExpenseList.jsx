import React from 'react';
import { connect } from 'react-redux';
import ExpenseListItem from './ExpenseListItem.jsx';
import selectExpenses from '../selectors/expenses';

export const ExpenseList = ({ expenses }) => (
  <div className="content-container">
    <div className="list-header">
      <div className="show-for-mobile">Expenses</div>
      <div className="show-for-desktop">Expense</div>
      <div className="show-for-desktop">Amount</div>
    </div>
    <div className="list-body">
      {expenses.length === 0 ? (
        <div className="list-item list-item--message">
          <span>No expenses</span>
        </div>
      ) : (
        expenses.map(({ id, description, amount, createdAt }) => (
          <ExpenseListItem
            key={id}
            id={id}
            description={description}
            amount={amount}
            createdAt={createdAt}
          />
        ))
      )}
    </div>
  </div>
);

const mapStateToProps = state => {
  return {
    expenses: selectExpenses(state.expenses, state.filters)
  };
};

export default connect(mapStateToProps)(ExpenseList);
