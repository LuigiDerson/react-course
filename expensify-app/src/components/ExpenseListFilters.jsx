/* eslint-disable */
import React from 'react';
import { connect } from 'react-redux';
import { DateRangePicker } from 'react-dates';
import { setTextFilter, sortByAmount, sortByDate, setStartDate, setEndDate } from '../actions/filters';

class ExpenseListFilters extends React.Component {
    state = {
        calendarFocused: null,
    }

    onDatesChange = ({ startDate, endDate }) => {
        this.props.dispatch(setStartDate(startDate));
        this.props.dispatch(setEndDate(endDate));
    }

    onFocusChange = (calendarFocused) => {
        this.setState(() => ({ calendarFocused }))
    }

    render() {
        return (
            <div>
                <input
                    type="text"
                    value={this.props.filters.text}
                    onChange={({target: {value: text}}) => {
                        this.props.dispatch(setTextFilter(text));
                }} />
                <select
                    value={this.props.filters.sortBy}
                    onChange={({target: {value: option}}) => {
                        if(option === 'date') {
                            this.props.dispatch(sortByDate());
                        } else if (option === 'amount') {
                            this.props.dispatch(sortByAmount());
                        }
                    }}
                >
                    <option value="date">Date</option>
                    <option value="amount">Amount</option>
                </select>
                <DateRangePicker
                    startDate={this.props.filters.startDate}
                    endDate={this.props.filters.endDate}
                    onDatesChange={this.onDatesChange}
                    focusedInput={this.state.calendarFocused}
                    onFocusChange={this.onFocusChange}
                    showClearDates={true}
                    numberOfMonths={1}
                    isOutsideRange={() => false}
                    displayFormat={"DD/MM/YYYY"}
                />
            </div>
        );
    }
};

const mapStatetoProps = (state) => ({
    filters: state.filters
});

export default connect(mapStatetoProps)(ExpenseListFilters);