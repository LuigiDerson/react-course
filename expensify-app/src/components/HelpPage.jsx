import React from 'react';

const HelpPage = () => (
    <div>
        This is from my help page Component
    </div>
);

export default HelpPage;