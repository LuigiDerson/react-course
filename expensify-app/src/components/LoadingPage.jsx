import React from 'react';

const LoadingPage = () => (
  <div className="loader">
    <div className="loader__image">
      <img src="/images/loader.gif" alt="loader-gif" />
    </div>
  </div>
);

export default LoadingPage;
