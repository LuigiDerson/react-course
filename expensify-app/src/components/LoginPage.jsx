import React from 'react';
import { connect } from 'react-redux';
import { startLogin } from '../actions/auth';

export const LoginPage = ({ startLoginAction }) => (
  <div className="box-layout">
    <div className="box-layout__box">
      <div>
        <h1 className="box-layout__title">Expensify App</h1>
        <p className="box-layout__subtitle">
          It&apos;s time to get your expenses under control
        </p>
        <button className="button" type="button" onClick={startLoginAction}>
          Login
        </button>
      </div>
    </div>
  </div>
);

const mapDispatchToProps = dispatch => ({
  startLoginAction: () => dispatch(startLogin())
});

export default connect(undefined, mapDispatchToProps)(LoginPage);
