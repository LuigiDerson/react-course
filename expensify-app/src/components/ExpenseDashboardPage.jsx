import React from 'react';
import ExpenseList from './ExpenseList.jsx';
import ExpenseListFilters from '../components/ExpenseListFilters.jsx';
import ExpensesSummary from './ExpensesSummary.jsx';

const ExpenseDashboardPage = () => (
    <div>
        <ExpensesSummary />
        <ExpenseListFilters />
        <ExpenseList />
    </div>
);

export default ExpenseDashboardPage;
