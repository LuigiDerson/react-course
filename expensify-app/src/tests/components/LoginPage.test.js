import React from 'react';
import { shallow } from 'enzyme';
import { LoginPage } from '../../components/LoginPage';

test('should correctly render LoginPage', () => {
  const wrapper = shallow(<LoginPage />);
  expect(wrapper).toMatchSnapshot();
});

test('should call startLogin on button click', () => {
  const startLoginAction = jest.fn();
  const wrapper = shallow(<LoginPage startLoginAction={startLoginAction} />);
  wrapper.find('button').simulate('click');
  expect(startLoginAction).toHaveBeenCalled();
});
