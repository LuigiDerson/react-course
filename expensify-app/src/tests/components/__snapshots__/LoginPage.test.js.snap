// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render LoginPage 1`] = `
<div
  className="box-layout"
>
  <div
    className="box-layout__box"
  >
    <div>
      <h1
        className="box-layout__title"
      >
        Expensify App
      </h1>
      <p
        className="box-layout__subtitle"
      >
        It's time to get your expenses under control
      </p>
      <button
        className="button"
        type="button"
      >
        Login
      </button>
    </div>
  </div>
</div>
`;
